### PR TITLE
Add a "Setup Python" action for github-hosted runners and remove unnecessary `CUSTOM_MINGW` environment variable

### PR DESCRIPTION
**Pull request by ChrisDenton**
The Setup Python action isn't strictly necessary ([even on Windows](https://github.com/rust-lang/rust/pull/125584)) but it is [recommend by GitHub](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#specifying-a-python-version).

The `CUSTOM_MINGW` environment variable is redundant now as it's always set for mingw and always unset otherwise.

try-job: x86_64-mingw
try-job: x86_64-mingw
try-job: dist-x86_64-msvc
try-job: dist-x86_64-mingw

### CODE REVIEW

**Review by mati865 — APPROVED** [review]

File: src/ci/scripts/install-mingw.sh [hunk 1]
```diff
@@ -1,8 +1,6 @@
 #!/bin/bash
 # If we need to download a custom MinGW, do so here and set the path
```
Comment (mati865) [inline src/ci/scripts/install-mingw.sh:2]: This comment could be updated.

Comment (ChrisDenton) [reply src/ci/scripts/install-mingw.sh]: Updated!

**Review by mati865 — APPROVED** [review]

**Comment by Kobzol** [comment]
Just to make sure that I understand the context, a custom MinGW was needed in the past, because the GitHub Windows runners had too old OS? Or is the reason that it is not needed now the fact that we have updated our minimum supported Windows versions?

**Comment by ChrisDenton** [comment]
> Just to make sure that I understand the context, a custom MinGW was needed in the past, because the GitHub Windows runners had too old OS? Or is the reason that it is not needed now the fact that we have updated our minimum supported Windows versions?

The windows-gnu runners do need a custom mingw to build rustc but they always need it so there's no reason to make it configurable via an environment variable. The script that uses `CUSTOM_MINGW` already has `if isWindows && isKnownToBeMingwBuild` so at that point `CUSTOM_MINGW` will always be set.

The windows-msvc runners don't need mingw to build rust as they use MSVC and LLVM tools. I'm not entirely clear why we used to install gnu build tools on windows-msvc runners but presumably it was needed at some point in the past. CI does need a way to run bash scripts and support for run-make tests (e.g. having a gnu `make`) but we don't need any special scripting to support that. It's already provided.

**Comment by Kobzol** [comment]
Looks like it works.

@bors r+

**Comment by Kobzol** [comment]
The setup Python action doesn't work for `arm64` yet, which is needed for https://github.com/rust-lang/rust/pull/126113. @ChrisDenton Do you have anything against removing it? I don't think that it provides any value to us, Ubuntu 20.04 comes with Python 3.8, and that should be more than enough for the few scripts that we have (at least, I don't think that this was a problem in practice in the past few years).

**Comment by ChrisDenton** [comment]
Seems fine to remove it. I don't think there's any particular reason we need it.

**Comment by Kobzol** [comment]
Ok, posted https://github.com/rust-lang/rust/pull/126116.

**Pull request merged**